### PR TITLE
Add golang env variables to setup documentation

### PR DESCRIPTION
**Pull request by bojbrook**
### What does this PR do?
Adds the go tracer environment variables to the configuration page

### Motivation
Jira issue: https://datadoghq.atlassian.net/browse/APMS-6429

### Preview
https://docs-staging.datadoghq.com/bowen/golang-env/tracing/setup_overview/setup/go/?tab=containers#configuration

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/setup_overview/setup/go.md [hunk 1]
```diff
@@ -61,6 +61,39 @@ Datadog has a series of pluggable packages which provide out-of-the-box support
...
+: **Default**: `localhost` <br>
+Override the address of the trace Agent host that the default tracer attempts to submit traces to.
+
+`DD_DOGSTATSD_PORT`
+: **Default**: `8125` <br>
+Override the port that the default tracer submits DogStatsD metrics to.
+
+`DD_TAGS`
+: **Default**: [] <br>
+A list of default tags to be added to every span and profile, for example: layer:api,team:intake
```
Comment (knusbaum) [inline content/en/tracing/setup_overview/setup/go.md:83]: We should specify here that this list may be *either* space-separated or comma-separated.
Space-separated tags are compatible with DD_TAGS in the agent, so is recommended. Comma-separation support is present for backwards-compatibility.

**Review by knusbaum — CHANGES_REQUESTED** [review]
Looks good except one note we should add.

File: content/en/tracing/setup_overview/setup/go.md (hunk 1 above)
Comment (bojbrook) [reply content/en/tracing/setup_overview/setup/go.md]: 👍

**Review by knusbaum — APPROVED** [review]

File: content/en/tracing/setup_overview/setup/go.md [hunk 2]
```diff
@@ -61,6 +61,40 @@ Datadog has a series of pluggable packages which provide out-of-the-box support
...
+
+`DD_SERVICE`
+: The service name to be used for this application. 
+
+`DD_ENV`
+: Set the application’s environment, for example: prod, pre-prod, staging.
+
+`DD_AGENT_HOST`
+: **Default**: `localhost` <br>
+Override the address of the trace Agent host that the default tracer attempts to submit traces to.
```
Comment (ruthnaebeck) [inline content/en/tracing/setup_overview/setup/go.md:75]:
```suggestion
Override the default trace Agent host address for trace submission.
```

File: content/en/tracing/setup_overview/setup/go.md [hunk 3]
```diff
@@ -61,6 +61,40 @@ Datadog has a series of pluggable packages which provide out-of-the-box support
...
+
+`DD_DOGSTATSD_PORT`
+: **Default**: `8125` <br>
+Override the port that the default tracer submits DogStatsD metrics to.
```
Comment (ruthnaebeck) [inline content/en/tracing/setup_overview/setup/go.md:79]:
```suggestion
Override the default trace Agent port for DogStatsD metric submission.
```

File: content/en/tracing/setup_overview/setup/go.md [hunk 4]
```diff
@@ -61,6 +61,40 @@ Datadog has a series of pluggable packages which provide out-of-the-box support
...
+
+`DD_TAGS`
+: **Default**: [] <br>
+A list of default tags to be added to every span and profile. Tags can be separated by commas or spaces, for example: `layer:api,team:intake` or `layer:api team:intake`
+
+`DD_TRACE_STARTUP_LOGS`
+: **Default**: `true` <br>
+Enable startup configuration and diagnostic log
```
Comment (ruthnaebeck) [inline content/en/tracing/setup_overview/setup/go.md:87]:
```suggestion
Enable startup configuration and the diagnostic log.
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]
Docs review

**Review by ruthnaebeck — APPROVED** [review]

**Pull request merged**